### ISSUE TITLE
Add more languages as map-languages

### ISSUE THREAD

**Issue by verdulo** [osmandapp/Osmand#2977]
Please consider adding the following languages as map-languages:

[Kashubian](https://en.wikipedia.org/wiki/Kashubian_language) (csb) - [419 translated names](https://taginfo.openstreetmap.org/keys/?key=name:csb), it's recognised as "regional language" in Poland, even some paper-maps shows Kashubian names.

[Silesian](https://en.wikipedia.org/wiki/Silesian_language) (szl) - it's rather micro-language, or dialect of Polish language, there's already [421 translated names](https://taginfo.openstreetmap.org/keys/?key=name:szl).

Constructed languages: [Ido](https://en.wikipedia.org/wiki/Ido_(language)) (io) and [Interlingua](https://en.wikipedia.org/wiki/Interlingua) (ia), not as popular as Esperanto, 318 (ido) and 303 (ia) unique names are translated - mostly names of countries, seas, and bigger cities.

It's not much but some languages with even less number of names are selectable in OsmAnd.

**Comment by vshcherb** [comment]
I'm wondering, do we need to support micro languages which are barely supported by OSM community. I think we should keep only languages which have at least 1 city full coverage. For any other reason it will be always a mix of 2-3 languages and hardly makes sense.

@sonora  Please note it is not about "display language" so here criterias are different. I would even vote to remove artificial languages cause it is not logical to me.  Map languages should be the language that local people (any group) to use names and it is not about the language these people speak.

**Comment by sonora** [comment]
@vshcherb : Victor, Good point.

I have started thinking about this a few weeks ago when we had almost a "hype" of new languages appearing after I had enabled regional support, (and my activities were primarily targeted at major features like en_GB or es_AR.)

Then we started having duscussions about display languages, and (even though there still are non-OsmAnd  related issues with ISO692 3 letter locale codes in Android), I have for myself decided to honor translators' efforts by adding languages with >10% translation rate as "incomplete" display languages in OsmAnd, and drop the "incomplete" label for >80%.

Map language is another story entirely. There is no such factor as "translation rate", measuring how much individual contributors are willing to invest. Map language occurrences will almost entirely be restricted to map objects in certain regional areas, and as such I find no fair measurement. I have resorted to adding languages to our OsmAnd  "map language" menu unconditionally, but I am aware this is very debatable, and will always only benefit
- only a very minute user community
- and only in a very limited geographical area

Long story short: My opinion is we
- stay with the 10%/80% rule for display languages
- but include languages as map language if an area exists in OSM where objects tend to be tagged in that language

Not a firm opinion I have here, but I have also not seen much resistance, or negative impact on our app?

**Comment by homersimpsons** [comment]
Can't you modify map files to include only local map language, and propose additional files to get others languages.

**Comment by verdulo** [comment]
The "map language" option is mostly for displaying names of countries and seas, because all nearby small objects such as streets, shops, monuments have only local names. Will you add the mentioned languages into OsmAnd?

**Comment by vshcherb** [comment]
I think  the feature
``` The "map language" option is mostly for displaying names of countries and seas```
is already in place cause we use "Display Language" for basemap and for Wikipedia objects (not a Map Language)

So in summary we should start cleaning map languages but not adding.

**Comment by verdulo** [comment]
@vshcherb Incorrect, I've already checked and "Map language" is for objects on the basemap (and regional maps), but "Display language" determines names in "Download map list"; Wikipedia POIs are not working in newest development releases. What is the problem with adding more languages?
